aggregations-and-analytics/09/09_cdc_materialized_view.md

```diff
@@ -0,0 +1,71 @@
+
+In this example, you'll monitor a table with insurance claim data related to animal attacks in Australia, and use Flink SQL to maintain an aggregated **materialized view** that is **incrementally updated** with the latest claim costs. You can find a different version of this example deploying Debezium, Kafka and Kafka Connect in [this repository](https://github.com/morsapaes/flink-sql-CDC).
+
+## Pre-requisites
+
+You'll need a running Postgres service to follow this example, so we bundled everything up in a `docker-compose` script to keep it self-contained. The only pre-requisite is to have [Docker](https://docs.docker.com/get-docker/) installed on your machine. :whale:
+
+To get the setup up and running, run:
+
+`docker-compose build`
+
+`docker-compose up -d`
+
+Once all the services are up, you can start the Flink SQL client:
+
+`docker-compose exec sql-client ./sql-client.sh`
+
+## How it Works
+
+The source table (`accident_claims`) is backed by the [`Flink CDC Postgres` connector](https://github.com/ververica/flink-cdc-connectors/wiki/Postgres-CDC-Connector), which reads the transaction log of the `postgres` database to continuously produce change events. So, whenever there is an `INSERT`, `UPDATE` or `DELETE` operation in the `claims.accident_claims` table, it will be propagated to Flink.
+
+```sql
+CREATE TABLE accident_claims (
+    claim_id INT,
+    claim_total FLOAT,
+    claim_total_receipt VARCHAR(50),
+    claim_currency VARCHAR(3),
+    member_id INT,
+    accident_date VARCHAR(20),
+    accident_type VARCHAR(20),
+    accident_detail VARCHAR(20),
+    claim_date VARCHAR(20),
+    claim_status VARCHAR(10),
+    ts_created VARCHAR(20),
+    ts_updated VARCHAR(20)
+) WITH (
+  'connector' = 'postgres-cdc',
+  'hostname' = 'postgres',
+  'port' = '5432',
+  'username' = 'postgres',
+  'password' = 'postgres',
+  'database-name' = 'postgres',
+  'schema-name' = 'claims',
+  'table-name' = 'accident_claims'
+ );
+```
+
+After creating the changelog table, you can query it to find out the aggregated insurance costs of all cleared claims per animal type (`accident_detail`):
+
+```sql
+SELECT accident_detail,
+       SUM(claim_total) AS agg_claim_costs
+FROM accident_claims
+WHERE claim_status <> 'DENIED'
+GROUP BY accident_detail;
+```
+
+How can you check that the CDC functionality is _actually_ working? The `docker` directory also includes a data generator script (`postgres_datagen.sql`) with a series of `INSERT` operations with new insurance claims; if you run it, you can see how the query results update in (near) real-time:
+
+`cat ./postgres_datagen.sql | docker exec -i flink-cdc-postgres psql -U postgres -d postgres`
+
```

Comment:
I feel like this is missing one wrap-up sentence. I need to think about it. There should be a conclusion and call to action.

Reply:
Agreed. Will add it in!